plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/MockThriftHttpMetastoreServer.java

```diff
@@ -0,0 +1,1763 @@
+    private int port;
+    public static final String httpPath = "/mock-end-point";
+
+    public MockThriftHttpMetastoreServer(InMemoryThriftMetastore delegate, TestHttpRequestInterceptor requestInterceptor)
+            throws Exception
+    {
+        ExecutorService executorService = Executors.newFixedThreadPool(10);
+        ExecutorThreadPool threadPool = new ExecutorThreadPool((ThreadPoolExecutor) executorService);
+
+        jettyServer = new Server(threadPool);
+        jettyServer.setStopAtShutdown(true);
+
+        ServerConnector connector = new ServerConnector(jettyServer, new HttpConnectionFactory());
+        port = getRandomAvailablePort();
+        connector.setPort(port);
+        connector.setReuseAddress(true);
+        jettyServer.addConnector(connector);
+
+        MockThriftHiveMetastoreHandler mockThriftHandler = new MockThriftHiveMetastoreHandler(delegate);
+        TProcessor processor = new ThriftHiveMetastore.Processor<>(mockThriftHandler);
+        thriftHttpServlet = new MockThriftHttpServelet(processor, new TBinaryProtocol.Factory(), Optional.of(requestInterceptor));
+        ServletContextHandler context = new ServletContextHandler(
+                ServletContextHandler.NO_SECURITY | ServletContextHandler.NO_SESSIONS);
+        jettyServer.setHandler(context);
+
+        context.addServlet(new ServletHolder(thriftHttpServlet), httpPath);
+        start();
+    }
+
+    public void start()
+            throws Exception
+    {
+        jettyServer.start();
+    }
+
+    public void stop()
+            throws Exception
+    {
+        jettyServer.stop();
+    }
+
+    @ResourcePresence
+    public boolean isRunning()
+    {
+        return jettyServer.isRunning();
+    }
+
+    private int getRandomAvailablePort()
+            throws IOException
+    {
+        int freePort = -1;
+        try (ServerSocket temp = new ServerSocket(0)) {
+            freePort = temp.getLocalPort();
+        }
+        return freePort;
+    }
+
+    public int getPort()
+    {
+        return port;
+    }
+
+    private static class MockThriftHttpServelet
+            extends TServlet
+    {
+        private final Optional<TestHttpRequestInterceptor> requestInterceptor;
+
+        public MockThriftHttpServelet(
+                TProcessor processor,
+                TProtocolFactory protocolFactory,
+                Optional<TestHttpRequestInterceptor> requestInterceptor)
+        {
+            super(processor, protocolFactory);
+            this.requestInterceptor = requestInterceptor;
+        }
+
+        @Override
+        protected void doPost(HttpServletRequest request,
+                HttpServletResponse response)
+                throws ServletException, IOException
+        {
+            requestInterceptor.ifPresent(testHttpRequestInterceptor -> testHttpRequestInterceptor.process(request));
+            super.doPost(request, response);
+        }
+    }
+
+    interface TestHttpRequestInterceptor
```

Comment:
This could be replaced with `Consumer< HttpServletRequest>`

Reply:
Done.